packages/@aws-cdk/aws-lambda-nodejs/README.md

```diff
@@ -0,0 +1,56 @@
+## Amazon Lambda Node.js Library
+<!--BEGIN STABILITY BANNER-->
+
+---
+
+![Stability: Experimental](https://img.shields.io/badge/stability-Experimental-important.svg?style=for-the-badge)
+
+> **This is a _developer preview_ (public beta) module. Releases might lack important features and might have
+> future breaking changes.**
+>
+> This API is still under active development and subject to non-backward
+> compatible changes or removal in any future version. Use of the API is not recommended in production
+> environments. Experimental APIs are not subject to the Semantic Versioning model.
+
+---
+<!--END STABILITY BANNER-->
+
+This library provides constructs for Node.js Lambda functions.
+
+### Node.js Function
+Define a `NodejsFunction`:
+
+```ts
+new lambda.NodejsFunction(this, 'my-handler');
+```
+
+By default, the construct will look up for a `index.ts` or `index.js` that exports a `handler`
+function in the folder named as the construct's id:
+```
+.
+├── my-handler
+|   └── index.ts # exports a function named 'handler'
+├── stack.ts # defines a 'NodeJsFunction' with 'my-handler' as id
+```
+
+This file is used as "entry" for [Parcel](https://parceljs.org/). This means that your code is
+automatically transpiled and bundled whether it's written in JavaScript or TypeScript.
+
+Alternatively, an entry file and handler can be specified:
+```ts
+new lambda.NodejsFunction(this, 'MyFunction', {
+  entry: '/path/to/my/file.ts'
+  handler: 'myExportedFunc'
+});
+```
+
+All other properties of `lambda.Function` are supported, see also the [AWS Lambda construct library](https://github.com/aws/aws-cdk/tree/master/packages/%40aws-cdk/aws-lambda).
+
+### Configuring Parcel
+The `NodejsFunction` construct exposes some Parcel options via properties: `minify`, `sourceMaps`,
+`buildDir` and `cacheDir`.
+
+Parcel transpiles your code (every internal module) with `@babel/preset-env` and uses the
+`engines.node` defined in `package.json` as target (defaults to Node.js 8).
```

Comment:
The default node version should be read from `process.versions.node` instead of `package.json`.

Reply:
Not sure I can tell Parcel to do that via CLI

Reply:
Not possible to specify the babel target via CLI. This just means that the code gets  transpiled "too much" if nothing is specified in `engines.node`.

An alternative would be to save the `package.json`, update it with a `engines.node` key and then revert it...

Reply:
It would be nice to tell parcel to compile to the node target specified in the function props `runtime` argument. Doesn't seem thats easily achievable though. This feels like the ideal over `process.versions.node` to me.

Reply:
Found a solution for this, see latest commit https://github.com/aws/aws-cdk/pull/5532/commits/24106894ff06361ad0060cd123cacbb1d3eead74.

@eladb @MrArnoldPalmer are you ok with modifying and then restoring a user's `package.json`?

Reply:
I am ok with it.